test/segment-transmuxer.test.js

```diff
@@ -0,0 +1,349 @@
+  },
+  afterEach(assert) {
+    if (this.transmuxer) {
+      this.transmuxer.terminate();
+    }
+  }
+});
+
+QUnit.test('transmux returns data for full appends', function(assert) {
+  const done = assert.async();
+  const dataFn = sinon.spy();
+  const trackInfoFn = sinon.spy();
+  const audioTimingFn = sinon.spy();
+  const videoTimingFn = sinon.spy();
+
+  this.transmuxer = this.createTransmuxer(false);
+
+  transmux({
+    transmuxer: this.transmuxer,
+    bytes: muxedSegment(),
+    audioAppendStart: null,
+    gopsToAlignWith: null,
+    isPartial: false,
+    onData: dataFn,
+    onTrackInfo: trackInfoFn,
+    onAudioTimingInfo: audioTimingFn,
+    onVideoTimingInfo: videoTimingFn,
+    onId3: this.noop,
+    onCaptions: this.noop,
+    onDone: () => {
+      assert.ok(dataFn.callCount, 'got data events');
+      assert.ok(trackInfoFn.callCount, 'got trackInfo events');
+      assert.ok(audioTimingFn.callCount, 'got audioTimingInfo events');
+      assert.ok(videoTimingFn.callCount, 'got videoTimingInfo events');
+      done();
+    }
+  });
+});
+
+QUnit.test('resets transmuxer on reset()', function(assert) {
+  this.transmuxer = this.mockTransmuxer(false);
+  this.transmuxer.postMessage = sinon.spy();
+
+  reset(this.transmuxer);
+  assert.deepEqual(
+    this.transmuxer.postMessage.args[0][0],
+    { action: 'reset' },
+    'called reset on transmuxer'
+  );
+});
+
+QUnit.test('passes endTimeline to transmuxer on endTimeline()', function(assert) {
+  this.transmuxer = this.mockTransmuxer(false);
+  this.transmuxer.postMessage = sinon.spy();
+
+  endTimeline(this.transmuxer);
+  assert.deepEqual(
+    this.transmuxer.postMessage.args[0][0],
+    { action: 'endTimeline' },
+    'called endTimeline on transmuxer'
+  );
+});
+
+QUnit.test('passes action to transmuxer on enqueueAction()', function(assert) {
+  this.transmuxer = this.mockTransmuxer(false);
+  this.transmuxer.postMessage = sinon.spy();
+
+  enqueueAction('push', this.transmuxer);
+  assert.deepEqual(
+    this.transmuxer.postMessage.args[0][0],
+    { action: 'push' },
+    'called push on transmuxer'
+  );
+  assert.deepEqual(
+    this.transmuxer.postMessage.callCount,
+    1,
+    'only posted one message'
+  );
+});
+
+QUnit.test('dequeues and processes action on dequeue()', function(assert) {
+  this.transmuxer = this.mockTransmuxer(false);
+  this.transmuxer.postMessage = sinon.spy();
+
+  assert.deepEqual(this.transmuxer.postMessage.callCount, 0, 'no actions yet');
+
+  transmux({
+    transmuxer: this.transmuxer,
+    bytes: new Uint8Array(),
+    audioAppendStart: null,
+    gopsToAlignWith: null,
+    isPartial: false,
+    onData: this.noop,
+    onTrackInfo: this.noop,
+    onAudioTimingInfo: this.noop,
+    onVideoTimingInfo: this.noop,
+    onId3: this.noop,
+    onCaptions: this.noop,
+    onDone: this.noop
+  });
+  enqueueAction('reset', this.transmuxer);
+  assert.deepEqual(this.transmuxer.postMessage.callCount, 1, 'reset is in the queue');
```

Comment:
Technically this doesn't check about reset specifically. Might be better to just say something like"one action posted" (can always have a comment above it too if we want)